controllers/datamover.go

```diff
@@ -160,3 +180,149 @@ func (r *DPAReconciler) getDataMoverLabels() map[string]string {
+	if resticsecret == nil {
+		return false
+	}
+	for key, val := range resticsecret.Data {
+		if key == ResticPassword {
+			if len(val) != 0 {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func (r *DPAReconciler) createResticSecretsPerBSL(bsl velerov1.BackupStorageLocation, resticsecretname string, pass []byte) (*corev1.Secret, error) {
+
+	switch bsl.Spec.Provider {
+	case AWSProvider:
+		{
+			secretName, secretKey := r.getSecretNameAndKey(&bsl.Spec, oadpv1alpha1.DefaultPluginAWS)
+			bslSecret, err := r.getProviderSecret(secretName)
+			if err != nil {
+				return nil, err
+			}
+
+			awsProfile := "default"
+
+			if value, exists := bsl.Spec.Config[Profile]; exists {
+				awsProfile = value
+			}
+
+			key, secret, err := r.parseAWSSecret(bslSecret, secretKey, awsProfile)
+			if err != nil {
+				r.Log.Info(fmt.Sprintf("Error parsing provider secret %s for backupstoragelocation", secretName))
+				return nil, err
+			}
+
+			repo := "s3:s3.amazonaws.com/" + bsl.Spec.ObjectStorage.Bucket
+
+			rsecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%s-resticsecret", bsl.Name),
+					Namespace: bsl.Namespace,
+					Labels: map[string]string{
+						"openshift.io/oadp-bsl-name": bsl.Name,
+					},
+				},
+			}
+
+			op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, rsecret, func() error {
+
+				if len(pass) == 0 {
+					return fmt.Errorf("restic password is empty")
+				}
+				rData := &corev1.Secret{
+					Data: map[string][]byte{
+						AWSAccessKey:     []byte(key),
+						AWSSecretKey:     []byte(secret),
+						AWSDefaultRegion: []byte(bsl.Spec.Config[Region]),
+						ResticPassword:   pass,
+						ResticRepository: []byte(repo),
+					},
+				}
+				rsecret.Data = rData.Data
+				return nil
+			})
+
+			if err != nil {
+				return nil, err
+			}
+			if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated {
+				r.EventRecorder.Event(rsecret,
+					corev1.EventTypeNormal,
+					"ResticSecretReconciled",
+					fmt.Sprintf("%s restic secret %s", op, rsecret.Name),
+				)
+			}
+		}
+		// TODO: Azure & GCP
+	}
+
+	return nil, nil
+}
+
+func (r *DPAReconciler) createResticSecret(dpa *oadpv1alpha1.DataProtectionApplication) (bool, error) {
+
+	// obtain restic secret name from the config
+	// if no name is mentioned in the config, ten assume default restic secret name
+	resticsecretname := "restic-secret"
+	cred := dpa.Spec.Features.DataMoverCredential
+	if cred != nil {
+		resticsecretname = cred.Name
+	}
+
+	// fetch restic secret from the cluster
+	resticSecret := corev1.Secret{}
+	if err := r.Get(r.Context, types.NamespacedName{Namespace: r.NamespacedName.Namespace, Name: resticsecretname}, &resticSecret); err != nil {
+		r.Log.Error(err, "unable to fetch Restic Secret")
+		return false, err
+	}
+
+	// validate restic secret
+	val := r.validateDataMoverCredential(&resticSecret)
+	if !val {
+		return false, fmt.Errorf("invalid restic credential")
```

Comment:
Since this only checks for a password, should the error reflect that? Something like "no password supplied in restic secret." wdyt?

Reply:
changed condition